AdGoBye.Plugins/IPlugin.cs

```diff
@@ -28,13 +27,39 @@ public interface IPlugin
     ///     Patch is the main entrypoint to a plugin's operations. Plugins are expected to carry out their respective
     ///     behaviours in this method.
     /// </summary>
-    /// <param name="contentId">A string representing the content's blueprint identifier</param>
-    /// <param name="dataDirectoryPath">The path of the cache directory that contains the __data file</param>
+    /// <param name="fileContainer">Container for the underlying asset being operated on</param>
+    /// <param name="dryRunRequested">Bool representing if the current operation is a dry run, you should only simluate changes if this is true</param>
     /// <returns>A <see cref="EPatchResult" /> that signifies the result of the plugin's patch operation</returns>
-    EPatchResult Patch(string contentId, string dataDirectoryPath);
+    EPatchResult Patch(ref ContentFileContainer fileContainer, bool dryRunRequested);
 
-    /// <param name="contentId">A string representing the content's blueprint identifier</param>
-    /// <param name="dataDirectoryPath">The path of the cache directory that contains the __data file</param>
-    /// <returns>A <see cref="EVerifyResult" /> that signifies the result of the plugin's patch operation</returns>
-    EVerifyResult Verify(string contentId, string dataDirectoryPath);
+    /// <summary>
+    ///     Verify is a non-edit stage where Plugins can run environment and validity checks on the asset before
+    ///     operating on it in <see cref="Patch"/>.
+    /// </summary>
+    /// <param name="fileContainer">Container for the underlying asset being operated on</param>
+    /// <returns>A <see cref="EVerifyResult" /> that signifies the result of the plugin's verify operation.
+    ///          Non-<see cref="EVerifyResult.Success"/> returns will skip this Plugin from being executed. </returns>
+    EVerifyResult Verify(ref readonly ContentFileContainer fileContainer);
+
+    /// <summary>
+    /// Initialize is a function that Plugins can use to set up their environment before executing
+    /// <see cref="Patch"/> and <see cref="Verify"/> functions.
+    /// </summary>
+    void Initialize();
+
+    /// <summary>
+    /// Initialize is a function that Plugins can use to clean up after themselves after execution.
```

Comment:
Documentation here isn't clear on when this actually occurs.

Also personally not a big fan of 'what should be done in the event' if it doesn't involve the functionality of the app itself. Might be best to separate that under a remark.
Is this after each patch, or after everything queued is done? Just specify that.

Reply:
Ditto of Initialize, ditto for the rephrasing I could think of:
```c#
/// PostPatch is an optional function ran after <see cref="Patch"/> in the patch loop which Plugins may use to clean
/// their state after patching a world. It's part of the patching loop and therefore may be called multiple times.
```

Reply:
Yea that looks good